workflow/util/util.go

```diff
@@ -866,6 +867,13 @@ func retryWorkflow(ctx context.Context, kubeClient kubernetes.Interface, hydrato
 	return wfClient.Update(ctx, newWF, metav1.UpdateOptions{})
 }
 
+func getTemplateFromNode(node wfv1.NodeStatus) string {
```

Comment:
Looks good! Could you add a test? In the meantime, I'll run it locally to test it out